notebooks/generate_annCSV.ipynb

```diff
@@ -0,0 +1 @@
 COCO(val_annot_path) # load annotations for validation set\n","\n","def get_meta(coco):\n","    ids = list(coco.imgs.keys())\n","    for i, img_id in enumerate(ids):\n","        img_meta = coco.imgs[img_id]\n","        ann_ids = coco.getAnnIds(imgIds=img_id)\n","        anns = coco.loadAnns(ann_ids)\n","        img_file_name = img_meta['file_name']\n","        w = img_meta['width']\n","        h = img_meta['height']\n","        url = img_meta['coco_url']\n","        \n","        yield [img_id, img_file_name, w, h, url, anns]\n","        \n","def convert_to_df(coco, data_set):\n","    images_data = []\n","    persons_data = []\n","    \n","    for img_id, img_fname, w, h, url, meta in get_meta(coco):\n","        images_data.append({\n","            'image_id': int(img_id),\n","            'src_set_image_id': int(img_id), # repeat id to reference after join\n","            'coco_url': url,\n","            'path': data_set + '/' + img_fname,\n","            'width': int(w),\n","            'height': int(h)\n","        })\n","        for m in meta: \n","            persons_data.append({\n","                'ann_id': m['id'],\n","                'image_id': m['image_id'],\n","                'is_crowd': m['iscrowd'],\n","                'bbox': m['bbox'],\n","                'area': m['area'],\n","                'num_keypoints': m['num_keypoints'],            \n","                'keypoints': m['keypoints'],\n","                'segmentation': m['segmentation']\n","            })\n","    \n","    images_df = pd.DataFrame(images_data) \n","    images_df.set_index('image_id', inplace=True)\n","\n","    persons_df = pd.DataFrame(persons_data) \n","    persons_df.set_index('image_id', inplace=True)\n","    \n","    return images_df, persons_df\n","\n","images_df, persons_df = convert_to_df(train_coco, 'train2017')     \n","train_coco_df = pd.merge(images_df, persons_df, right_index=True, left_index=True)\n","train_coco_df['source'] = 0\n","train_coco_df.head()\n","\n","images_df, persons_df = convert_to_df(val_coco, 'val2017')      \n","val_coco_df = pd.merge(images_df, persons_df, right_index=True, left_index=True)\n","val_coco_df['source'] = 1\n","val_coco_df.head()\n","\n","coco_df = pd.concat([train_coco_df, val_coco_df], ignore_index=True)\n","# ^ Dataframe containing all val and test keypoint annotations\n","\n","#cleaned_coco_df\n","cleaned_coco_df = coco_df.loc[coco_df['num_keypoints'] >3].reset_index()\n","cleaned_coco_df"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"xtUj7_dUuozM"},"source":["anns_per_img = coco_df.groupby('src_set_image_id',as_index=False).size()\n","imgs_w_one_ann = anns_per_img.loc[anns_per_img['size'] == 1]\n","imgs_w_one_ann = pd.merge(imgs_w_one_ann,coco_df)\n","one_ann_min10kp_imgs = imgs_w_one_ann.loc[imgs_w_one_ann['num_keypoints'] > 10]\n","one_ann_min10kp_imgs\n","csv_data = one_ann_min10kp_imgs.to_csv('data/one_ann_min10kp_imgs.csv',index=False)"],"execution_count":null,"outputs":[]}]}
```

Comment:
<p>I think if possible we could move this code to a python file, and then call that file from the colab notebook?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/robertklee/COCO-Human-Pose/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Reply:
<p>Yup good call. I'll make a module that can grab a df of all coco annotations. Then in a notebook we'll be able to filter for the anns we want before writing to csv.</p>

Reply:
yep. Ideally i think our `train.py` and `eval.py` should handle everything end-to-end so the end user needs to just clone, `python3 train.py <arguments>`, and then `python3 eval.py <arguments>`. 

Let's keep this in mind as we develop, and try to make things modular enough that we can implement this later on.

Reply:
Of course our Google Colab filesystem issues will make this not work lol